File: RealTime/Queries/OrderBy.cs
using System;
using System.Collections.Generic;

namespace QualityGate.RealTime.Queries
{
    /// <summary>
    ///     Contains order-by sorting definition.
    /// </summary>
    public record OrderBy
    {
        /// <summary>
        ///     Gets or sets the enumerable of names of the fields to order by.
        /// </summary>
        public IEnumerable<string>? Fields { get; set; } = Array.Empty<string>();

        /// <summary>
        ///     Gets or sets a boolean saying whether the ordering is ascending (if true) or descending.
        /// </summary>
        public bool Ascending { get; set; } = true;
    }
}

File: Npm/contracts/queries/OrderBy.ts
/**
 * Defines ordering in a query. Says by which fields to order and by wish fashion (ascending or descending).
 */
export interface OrderBy {
	/**
	 * Sets the names of the field to order the query results by. The order matters.
	 */
	fields: string[]

	/**
	 * Sets whether the query results must be ordered ascending or descending.
	 */
	ascending: boolean
}